### PR TITLE
Support GHC 9.0

### PR DESCRIPTION
**Pull request by ghost**
- [x] Get all projects compiling, tests passing
- [ ] Remove packages from brokenlist in stackage: fpco/stackage#6072

### CODE REVIEW

**Comment by ghost** [comment]
Thank you for driving this forward. It's working great! I got one idea for pushing the ghc-version-dependent nix logic into the `shell-ghc-X.nix` files. Going to give that a try and then merge.

File: README.md [hunk 1]
```diff
@@ -30,11 +30,11 @@ Documentation comments use the [Haddock][haddock] format to ensure they are rend
 
 ### Testing compilation against GHC 8.10
 
-Our default `shell.nix` sets up an environment around GHC version 8.10. CI also runs tests against GHC 8.8. To test using this version locally:
+Our default `shell.nix` sets up an environment around GHC version 8.10. CI also runs tests against GHC 9.0. To test using this version locally:
```
Comment (ghost) [inline README.md:33]: :heart_eyes_cat:

File: .travis.yml [hunk 2]
```diff
@@ -2,6 +2,9 @@ language: nix
 nix: 2.3.7
 os: linux
 dist: xenial
+vm:
+  # we were getting out of memory errors without this
+  size: large
```
Comment (ghost) [inline .travis.yml:7]: :heart_eyes_cat:

File: shell-ghc-9-0.nix [hunk 3]
```diff
@@ -4,7 +4,10 @@ let
   pkgs = import sources.nixpkgs-unstable { };
 in import nix/mk-shell.nix {
   pkgs = pkgs;
-  haskellPackages = pkgs.haskell.packages.ghc901;
-  useServant_0_18_3 = true;
-  customSafeColouredText = false;
+  haskellPackages = pkgs.haskell.packages.ghc901.extend (self: super: {
```
Comment (michaelglass) [inline shell-ghc-9-0.nix:7]: much nicer. I tried this approach initially but gave up "trying to fix too many problems at once"

File: .travis.yml [hunk 4]
```diff
@@ -9,11 +9,13 @@ env:
   global:
     - PGPORT=8088
     - CACHIX_CACHE=nri
+  jobs:
```
Comment (michaelglass) [inline .travis.yml:12]: 🤘

**Review by michaelglass — APPROVED** [review]

File: nri-redis/src/Redis/Internal.hs [hunk 5]
```diff
@@ -198,7 +198,7 @@ query :: Stack.HasCallStack => Handler -> Query a -> Task Error a
 query handler query' =
   namespaceQuery (namespace handler ++ ":") query'
     |> Task.andThen (ensureMaxKeySize handler)
-    |> Task.andThen (Stack.withFrozenCallStack doQuery handler)
+    |> Task.andThen (Stack.withFrozenCallStack (doQuery handler))
```
Comment (michaelglass) [inline nri-redis/src/Redis/Internal.hs:201]: I don't totally understand `Stack.withFrozenCallStack` or execution order here (or really, deep skolemisation in general) 

But
- do we want to do this instead? Does it make a difference?
```suggestion
    |> Task.andThen (\query -> Stack.withFrozenCallStack ((doQuery handler) query))
```

Comment (ghost) [reply nri-redis/src/Redis/Internal.hs:201]: I don't think we want to do this. My partial understanding is that doing the above would prevent `withFrozenCallStack` from doing something useful (I've seen that happen in the past, though not  consistently).

The way I look at this (which might be wrong): Currently `withFrozenCallStack` is being passed a function, which makes sense because we want to modify that function's behavior to stop adding stack frames.

With the change you propose we'd be passing `withFrozenCallStack` a value that is the result of calling the function. i.e., by the time `withFrozenCallStack` is called, the function it wants to change the behavior of already ran, and we're too late.

Maybe our historical problems using `withFrozenCallStack` are related to use running strictly by default, and lazily evaluating functions would make it so `withFrozenCallStack` would work in the way you propose here.

Bottom line is I don't really understand how it works either. It's annoying and super easy to break accidentally in ways that makes our loggers start registering useless internal source locations.

I believe we have decent test coverage for this stuff though, so a "as long as the tests are happy" approach seems fair to me as well.

**Comment by ghost** [comment]
Oops, should have done version increments before merging.

Going to create a separate PR for that.
